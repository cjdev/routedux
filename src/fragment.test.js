import React from 'react';
import Fragment from './fragment';
import ezJson from 'enzyme-to-json';
import {shallow} from 'enzyme';


it("should display when state is truthy", () => {

  // given
  const state = {property : true};
  // when

  const wrapper = shallow(<Fragment state={state} filterOn="property">
      <div>Hello</div>
  </Fragment>);

  // then
  expect(ezJson(wrapper)).toMatchSnapshot();
});

it("should not display when state is falsy", () => {
  // given
  const state = {property : undefined};
  // when

  const wrapper = shallow(<Fragment state={state} filterOn="property">
    <div>Hello</div>
  </Fragment>);

  // then
  expect(ezJson(wrapper)).toEqual(null);
});

it("should handle paths in the state tree", () => {
  // given
  const state = {property : {subproperty: true}};
  // when

  const wrapper = shallow(<Fragment state={state} filterOn="property.subproperty">
    <div>Hello</div>
  </Fragment>);

  // then
  expect(ezJson(wrapper)).toMatchSnapshot();
});

it("should handle arrays in the state tree", () => {
  // given
  const state = {property : [{bar: {}}]};
  // when

  const wrapper = shallow(<Fragment state={state} filterOn="property.0.bar">
    <div>Hello</div>
  </Fragment>);

  // then
  expect(ezJson(wrapper)).toMatchSnapshot();
});

it("should be falsy if missing state tree", () => {
  // given
  const state = {property : {subproperty: true}};

  const wrapper = shallow(<Fragment state={state} filterOn="property.missingproperty.something">
    <div>Hello</div>
  </Fragment>);

  expect(ezJson(wrapper)).toEqual(null);

});
