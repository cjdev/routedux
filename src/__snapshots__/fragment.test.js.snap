// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display when state is truthy 1`] = `
<div>
  Hello
</div>
`;

exports[`should handle arrays in the state tree 1`] = `
<div>
  Hello
</div>
`;

exports[`should handle paths in the state tree 1`] = `
<div>
  Hello
</div>
`;
